src/openassetio-core/src/pluginSystem/HybridPluginSystemManagerImplementationFactory.cpp

```diff
@@ -0,0 +1,421 @@
+        if (managerInterface->hasCapability(capability)) {
+          managerInterfacesByCapability_[capability] = managerInterface;
+          break;
+        }
+      }
+    }
+  }
+
+  /**
+   * Return the identifier of the first child, on the understanding that
+   * all child implementations have the same identifier.
+   */
+  [[nodiscard]] Identifier identifier() const override {
+    return managerInterfaces_.front()->identifier();
+  }
+
+  /**
+   * Return the name of the first child, on the assumption that all
+   * child implementations have the same name.
+   */
+  [[nodiscard]] Str displayName() const override {
+    return managerInterfaces_.front()->displayName();
+  }
+
+  /**
+   * Merge the results of `info()` from all child implementations, with
+   * the first implementation taking precedence in case of a conflict.
+   */
+  [[nodiscard]] InfoDictionary info() override {
+    InfoDictionary result;
+    for (const auto& managerInterface : managerInterfaces_) {
+      result.merge(managerInterface->info());
+    }
+    return result;
+  }
+
+  /**
+   * Merge the results of `settings()` from all child implementations,
+   * with the first implementation taking precedence in case of a
+   * conflict.
+   */
+  [[nodiscard]] InfoDictionary settings(const managerApi::HostSessionPtr& hostSession) override {
+    InfoDictionary result;
+    for (const auto& managerInterface : managerInterfaces_) {
+      result.merge(managerInterface->settings(hostSession));
+    }
+    return result;
+  }
+
+  /**
+   * This hybrid manager has a capbility if at least one of its child
+   * implementations has the capability.
+   */
+  [[nodiscard]] bool hasCapability(const Capability capability) override {
+    return managerInterfacesByCapability_.count(capability) > 0;
+  }
+
+  /**
+   * All child implementations are initialized with the all the same
+   * settings.
+   */
+  void initialize(InfoDictionary managerSettings,
+                  const managerApi::HostSessionPtr& hostSession) override {
+    for (const auto& managerInterface : managerInterfaces_) {
+      managerInterface->initialize(managerSettings, hostSession);
+    }
+  }
+
+  /**
+   * All child implementations flushed.
+   */
+  void flushCaches(const managerApi::HostSessionPtr& hostSession) override {
+    for (const auto& managerInterface : managerInterfaces_) {
+      managerInterface->flushCaches(hostSession);
+    }
+  }
+
+  /*
+   * The remainining API methods all dispatch to the first child that
+   * has the capability, or to the base class implementation if no child
+   * has the capability.
+   */
+
+  [[nodiscard]] StrMap updateTerminology(StrMap terms,
+                                         const managerApi::HostSessionPtr& hostSession) override {
+    if (const auto& capabilityAndManager =
+            managerInterfacesByCapability_.find(Capability::kCustomTerminology);
+        capabilityAndManager != cend(managerInterfacesByCapability_)) {
```

Comment:
Good idea. I did a similar thing in 4899d0d4

Reply:
I'll admit I'm weirded out by a macro defining a lambda that gets invoked in that commit
```
#define INVOKE_CAPABLE_MANAGER_FOR_FUNCTION(capability, method, ...)                        \
  [&] {                                                                                     \
    if (const auto& capabilityAndManager = managerInterfacesByCapability_.find(capability); \
        capabilityAndManager != cend(managerInterfacesByCapability_)) {                     \
      return capabilityAndManager->second->method(__VA_ARGS__);                             \
    }                                                                                       \
    return ManagerInterface::method(__VA_ARGS__);                                           \
  }()
```

Is it not possible to have a templated function with a parameter pack for the varargs, that requires the method to be invoked, and the return type on the template types?

While I'm writing that, I'm thinking perhaps it's more work, but there's something that nags at me about macros, typesafety, and still weirded out by that lambda in a macro.

Reply:
I did start with a template function rather than a macro, but it gets complicated by the `->` vs. `::` depending whether we're calling one of the child implementations or the base class.  I started to implement some lengthy member function pointer casting and `std::bind` type stuff, before realising that a macro would be much more succinct.

Wrapping the macro contents in an immediately invoked lambda was for aesthetics, specifically so that the member functions that use the macro could `return` the result of the macro.  We could get rid of the lambda, but then the `return` statements disappear from inside the methods (because they're in the macro instead).  I had it like that originally and it just looked weird to have a function with a return value yet no `return` statement in the body of the function.

Reply:
Sounds like you already explored the options, so let's leave it :)